### PR TITLE
Feat/add google fonts

### PR DESCRIPTION
**Pull request by jtrbuhovic**
# Context
Adds a data file where users can define Google fonts to be used within the website. The website can then assign fonts to elements/content within the `site` datafile under `styles`.

[Link to Notion ticket](https://www.notion.so/cloudcannon/Google-Fonts-Integration-bc1df27314ca4e6487f536a787f516c4?pvs=4)

# Testing (tester)

The reviewer should check on this branch:

- Adding a Google font to `src/_data/fonts.yml`
- Changing a font within `Styles` in `src/_data/site.json`
- Check the fonts are being imported from Google fonts in the `<head>` tag, and check that the fonts are applied to text.

# Before merge (PR owner)

- Delete the test site in CloudCannon

### CODE REVIEW

**Review by Amber-Song — APPROVED** [review]
LGTM.

**Review by aleksandrgordeev — APPROVED** [review]
Nice addition

File: component-library/shared/styles/global.scss [hunk 1]
```diff
@@ -2,7 +2,9 @@
 // as well as in the component browser and screenshots.
 
 html, body {
-    
+	h1, h2, h3, h4 {
```
Comment (Yoda-Soda) [inline component-library/shared/styles/global.scss]: Think this code should be moved to assets style sheet. I am also wondering about the code below this. This .component {} not sure if this is needed.

**Review by Yoda-Soda — CHANGES_REQUESTED** [review]

File: component-library/shared/styles/global.scss (hunk 1 above)
Comment (jtrbuhovic) [reply component-library/shared/styles/global.scss]: Yep this was put in as example usage and can be removed. 

Do you have a suggestion of how else we would apply the background color variant to all components? I used a single class so it doesn't need to be re-implemented with each component. All components have a dark/light variant toggle so there needs to be a way for the user to define those colors, and then a simple way to apply them to each component based on the chosen variant. Using a class also allows you to add as many shared component styles as necessary

**Comment by James-cloudcannon** [comment]
Importing fonts works great but how do you plan on changing --heading-font and --content-font to match these updates as they currently stay as Lato? You could add a simple style tag into the head, that sets these css variables from the site.json file? such as:

// page.html or google-fonts.html
`<style>
  :root {
    --heading-font: {{ site.styles.heading_font }};
  }
</style>`

**Comment by jtrbuhovic** [comment]
> Importing fonts works great but how do you plan on changing --heading-font and --content-font to match these updates as they currently stay as Lato? You could add a simple style tag into the head, that sets these css variables from the site.json file? such as:
> 
> // page.html or google-fonts.html `<style> :root { --heading-font: {{ site.styles.heading_font }}; } </style>`

That's a good idea, I might do that instead. The reason I've done it differently is because you can't use template language in SCSS files with eleventy :(. Typically I would just do something like this which would work fine:

`site scss file:`
```
---
---
:root {
    --heading-font: {{ site.data.site.styles.heading_font }}
}
```
The current setup works but it requires a full rebuild locally since the Node script needs to turn these user options into variables (`/src/assets/js/generate-styles`). I think your solution is much better for a new dev to understand, and will work better in general. I'll just substitute the script with an inline style element.

File: component-library/shared/styles/global.scss (hunk 1 above)
Comment (jtrbuhovic) [reply component-library/shared/styles/global.scss]: I've made some improvements based on James' suggestion below

**Comment by jtrbuhovic** [comment]
@James-cloudcannon I've just made those changes 👍

**Comment by James-cloudcannon** [comment]
> @James-cloudcannon I've just made those changes 👍

Tested and works well. I am happy to merge this if @Yoda-Soda is fine with the requested changes.

File: component-library/shared/styles/global.scss (hunk 1 above)
Comment (jtrbuhovic) [reply component-library/shared/styles/global.scss]: @Yoda-Soda can you please check this again when possible

**Comment by AlyshaNolan** [comment]
Unless anyone can see a reason not to I'd like to change the google fonts partial to only include the two fonts selected in site.json (instead of everything in fonts.yml) as including multiple unused fonts can affect site performance

**Comment by jtrbuhovic** [comment]
I've just pushed those changes @AlyshaNolan, great suggestion 👍

**Review by AlyshaNolan — APPROVED** [review]

**Review by Yoda-Soda — APPROVED** [review]

**Pull request merged**